shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -253,12 +259,22 @@ private void setSystemChromeEnabledSystemUIMode(PlatformChannel.SystemUiMode sys
       // the system overlay has changed. The overlays cannot be dismissed, so adding the callback
       // support will allow users to restore the system ui and dismiss the overlays.
       // Not compatible with top/bottom overlays enabled.
-      enabledOverlays =
-          View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-              | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-              | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_FULLSCREEN;
+      if (Build.VERSION.SDK_INT >= 30) {
+        windowInsetsControllerCompat.setSystemBarsBehavior(
+            WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_TOUCH);
+        WindowCompat.setDecorFitsSystemWindows(window, false);
+        windowInsetsControllerCompat.hide(
+            WindowInsets.Type.statusBars() | WindowInsets.Type.navigationBars());
+
+        return;
+      } else {
+        enabledOverlays =
+            View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_FULLSCREEN;
```

Comment:
These flags are also deprecated aren't they? Or are these going to be migrated in a separate change?

Reply:
The new API is backwards compatible, so I will take care of these in this PR as well!